x-pack/plugins/task_manager/server/task_store.ts

```diff
@@ -504,6 +505,36 @@ export class TaskStore {
     }
   }
 
+  async msearch(opts: SearchOpts[] = []): Promise<FetchResult> {
+    const queries = opts.map((opt) => ensureQueryOnlyReturnsTaskObjects(opt));
+    const body = queries.flatMap((query) => [{}, query]);
+
+    const result = await this.esClientWithoutRetries.msearch<SavedObjectsRawDoc['_source']>({
+      index: this.index,
+      ignore_unavailable: true,
+      body,
+    });
+    const { responses } = result;
+
+    const versionMap = this.createVersionMap([]);
+    let allTasks = new Array<ConcreteTaskInstance>();
+
+    for (const response of responses) {
+      if (response.status !== 200) {
+        throw new Error(`Unexpected status code: ${response.status}`);
```

Comment:
should we pass this error to `this.errors$` as well?

Reply:
Ya, though it's making me wonder, with the weird partial result error stuff from CCS calls, should we just skip over these?  If just one of the queries is bad for some reason, but the other ones were ok, and that was consistent, we'd never pull any tasks.  Vs pulling tasks for everything but one of the "inner searches" failing.

I guess we'll figure that out ... :-)

Reply:
added in 842d4023a447ed91b81627faee668b0bcb2a1e16